ui/components/app/advanced-gas-fee-popover/advanced-gas-fee-inputs/base-fee-input/base-fee-input.js

```diff
@@ -0,0 +1,201 @@
+import React, { useCallback, useEffect, useState } from 'react';
+import { useSelector } from 'react-redux';
+
+import { HIGH_FEE_WARNING_MULTIPLIER } from '../../../../../pages/send/send.constants';
+import { PRIORITY_LEVELS } from '../../../../../../shared/constants/gas';
+import {
+  divideCurrencies,
+  multiplyCurrencies,
+} from '../../../../../../shared/modules/conversion.utils';
+import { PRIMARY, SECONDARY } from '../../../../../helpers/constants/common';
+import { bnGreaterThan, bnLessThan } from '../../../../../helpers/utils/util';
+import { decGWEIToHexWEI } from '../../../../../helpers/utils/conversions.util';
+import { getAdvancedGasFeeValues } from '../../../../../selectors';
+import { useGasFeeContext } from '../../../../../contexts/gasFee';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import { useUserPreferencedCurrency } from '../../../../../hooks/useUserPreferencedCurrency';
+import { useCurrencyDisplay } from '../../../../../hooks/useCurrencyDisplay';
+import Button from '../../../../ui/button';
+import Box from '../../../../ui/box';
+import FormField from '../../../../ui/form-field';
+import I18nValue from '../../../../ui/i18n-value';
+
+import { useAdvanceGasFeePopoverContext } from '../../context';
+import AdvancedGasFeeInputSubtext from '../../advanced-gas-fee-input-subtext';
+
+const divideCurrencyValues = (value, baseFee) => {
+  if (baseFee === 0) {
+    return 0;
+  }
+  return divideCurrencies(value, baseFee, {
+    numberOfDecimals: 2,
+    dividendBase: 10,
+    divisorBase: 10,
+  }).toNumber();
+};
+
+const multiplyCurrencyValues = (baseFee, value, numberOfDecimals) =>
+  multiplyCurrencies(baseFee, value, {
+    numberOfDecimals,
+    multiplicandBase: 10,
+    multiplierBase: 10,
+  }).toNumber();
+
+const validateBaseFee = (value, gasFeeEstimates, maxPriorityFeePerGas) => {
+  if (bnGreaterThan(maxPriorityFeePerGas, value)) {
+    return 'editGasMaxBaseFeeImbalance';
+  }
+  if (
+    gasFeeEstimates?.low &&
+    bnLessThan(value, gasFeeEstimates.low.suggestedMaxFeePerGas)
+  ) {
+    return 'editGasMaxBaseFeeLow';
+  }
+  if (
+    gasFeeEstimates?.high &&
+    bnGreaterThan(
+      value,
+      gasFeeEstimates.high.suggestedMaxFeePerGas * HIGH_FEE_WARNING_MULTIPLIER,
+    )
+  ) {
+    return 'editGasMaxBaseFeeHigh';
+  }
+  return undefined;
```

Comment:
Same as above, should this be `null` rather than `undefined`?

Reply:
fixed